manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml

```diff
@@ -41,6 +41,10 @@ rules:
     verbs: [ "get", "watch", "list", "update", "patch", "create", "delete" ]
     resources: [ "workloadentries/status" ]
 
+  - apiGroups: ["networking.istio.io"]
```

Comment:
Suggestion: create a separate role.

Also conditional grant this role to Istiod, using some helm value - not sure if we want to enable this by default, or require that each Istiod is the only supported allocator, etc. 

And if user doesn't enable autoalloc - or wants to use a different allocator - no need to have this permission here. I know we followed a (pretty bad) practice of granting huge permissions to Istiod unconditionally - but at least for new stuff we should do better, and this feature has very serious security implications ( combined with DNS programming ).

Reply:
I'm talking about update/patch in particular - get/watch/list is fine.